src/ekat/ekat_workspace_impl.hpp

```diff
@@ -486,6 +535,29 @@ void WorkspaceManager<T, D>::Workspace::release_many_contiguous(
   m_team.team_barrier();
 }
 
+template <typename T, typename D>
+template <typename S>
+KOKKOS_INLINE_FUNCTION
+void WorkspaceManager<T, D>::Workspace::release_n_size_block(
+  const Unmanaged<view_1d<S> >& space, const int n) const
+{
+#ifndef NDEBUG
+  change_num_used(-n);
```

Comment:
I think this is correct, we can't check metadata here since it may have been overwritten by application data.

Reply:
Ah. Are you saying we are returning something like nlev*sizeof(double)+room_for_metadata _per slot_, but since we're using all those slots as a single 2d array, we are writing actual data in the slots where usually we keep metadata? And are you saying this is not a pb, since the release method for the 2d array is not gonna check those?

Reply:
@bartgol , yes that's exactly what I mean.

Reply:
Side question: is it worth adding a check in get_index, that makes sure the input 1d view is "aligned", meaning that the user didn't accidentally pass in the wrong view? It would require checking that the `space.data()-m_data = K*m_size`, with 0<=0<m_max_used.

Or is it an overkill?